src/structure/IndexMap.lua

```diff
@@ -8,27 +10,43 @@ local IndexMap = class("IndexMap",{
 local Location = class("Location",{
 	address = nil,
 	placeables = {},
-	neighbors = {}
+	neighbors = {},
+	terrain_info = {}
 })
 
-function Location:init(address, neighbors)
+function Location:init(address, x, y, neighbors, terrain_info)
```

Comment:
I'd prefer not to have x,y needed here: I don't think you end up needing it since you've implemented your latest pathfinding by hand :)